### PR TITLE
Use count property instead of length

### LINKED ISSUE #3493: Reader: Recommendations: Console error: iterable.length has been deprecated, use iterable.size or iterable.count().
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/recommendations
2. View the console

#### What I expected

No errors/warnings

#### What happened instead

```
iterable.length has been deprecated, use iterable.size or iterable.count(). This warning will become a silent error in a future version. Error
    at OrderedSet.Object.defineProperty.get (https://wordpress.com/calypso/build-stage.f4ff76ac951d0b022036.min.js:8:22041)
    at receiveRecommendations (https://wordpress.com/calypso/reader.01a178ddf716600d270c.min.js:9:16760)
    at Object.ID_26 (https://wordpress.com/calypso/reader.01a178ddf716600d270c.min.js:9:17090)
    at Dispatcher._invokeCallback (https://wordpress.com/calypso/build-stage.f4ff76ac951d0b022036.min.js:40:10129)
    at Dispatcher.dispatch (https://wordpress.com/calypso/build-stage.f4ff76ac951d0b022036.min.js:40:9876)
    at Dispatcher.handleServerAction (https://wordpress.com/calypso/build-stage.f4ff76ac951d0b022036.min.js:2:10112)
    at https://wordpress.com/calypso/reader.01a178ddf716600d270c.min.js:12:12330
    at https://wordpress.com/calypso/reader.01a178ddf716600d270c.min.js:6:31293
    at https://wordpress.com/calypso/vendor.f4ff76ac951d0b022036.min.js:8:27316
    at XMLHttpRequest.onload (https://wordpress.com/calypso/build-stage.f4ff76ac951d0b022036.min.js:21:13660)
```

#### Browser / OS version

Chrome OSX 10.11.3

### PR DESCRIPTION
**Pull request by gyoshev**
Resolves immutable.js deprecation warning.

Closes #3493

### CODE REVIEW

File: client/lib/recommended-sites-store/store.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ function receiveRecommendations( data ) {
 	// consume the recommendations
 	const previousRecs = recommendations;
 	if ( data && data.blogs ) {
-		if ( data.blogs.length === 0 ) {
+		if ( data.blogs.count === 0 ) {
```
Comment (TooTallNate) [inline client/lib/recommended-sites-store/store.js]: I believe that `count` must be a function, i.e. `count()`.

Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: yup https://facebook.github.io/immutable-js/docs/#/Seq/count

Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: but the thing i don't understand is how data.blogs is becoming an immutable instance...

Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: ah, it's not. the warning is being thrown further down, when we check the length of recommedations and set last page.

File: client/lib/recommended-sites-store/store.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
 		recommendations = recommendations.union( fromJS( pruned ) );
 
 		if ( recommendations !== previousRecs ) {
-			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
+			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
```
Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: so this is the actual problem and should be `count()`

Comment (gyoshev) [inline client/lib/recommended-sites-store/store.js]: Ah, good catch. I'll address this shortly; do you find it sensible to add
this code branch to the test suite, as it currently passes the tests?

On Tuesday, February 23, 2016, Ben Lowery <notifications@github.com> wrote:

> In client/lib/recommended-sites-store/store.js
> <https://github.com/Automattic/wp-calypso/pull/3502#discussion_r53831251>:
>
> > @@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
> >  		recommendations = recommendations.union( fromJS( pruned ) );
> >
> >  		if ( recommendations !== previousRecs ) {
> > -			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
> > +			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
>
> so this is the actual problem and should be count()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wp-calypso/pull/3502/files#r53831251>.
>


-- 
Alexander A. Gyoshev
Web Developer / User Interface Engineer
mobile: +359 888 255838
Sofia, Bulgaria
http://gyoshev.net/

Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: Sure, we're pretty light on tests for this store right now. Any help would be appreciated. :)

**Comment by gyoshev** [comment]
I fixed the count call and opened up the store to make it more testable:
* exposed the maximum recommendations constant
* introduced a method to reset the store between tests

Now there are at least a few tests for the isLastPage condition.

Your feedback is welcome!

File: client/lib/recommended-sites-store/store.js [hunk 3]
```diff
@@ -5,13 +5,19 @@ import pick from 'lodash/pick';
...
-const MAX_RECOMMENDATIONS = 100;
-
 const store = {
+	_reset() {
+		recommendations = OrderedSet();
+		page = 1;
+		isLastPage = false;
+	},
+	maxRecommendations() {
+		return 100;
```
Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: Let's bump this over into `./constants`

**Comment by blowery** [comment]
Looks great, thanks! :ship:

File: client/lib/recommended-sites-store/store.js (hunk 3 above)
Comment (blowery) [inline client/lib/recommended-sites-store/store.js]: actually, it matches what we had and it's only used one place. Let it lie.

**Pull request merged**